src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -643,6 +643,27 @@ R"(#include <stdio.h>
                 src = "(bool)(" + src + ")";
                 break;
             }
+            case (ASR::cast_kindType::LogicalToReal) : {
+                src = "(float)(" + src + ")";
+                break;
+            }
+            case (ASR::cast_kindType::RealToLogical) : {
+                src = "(bool)(" + src + ")";
+                break;
+            }
+            case (ASR::cast_kindType::LogicalToCharacter) : {
+                src = "(char)(" + src + ")";
+                break;
+            }
+            case (ASR::cast_kindType::CharacterToLogical) : {
+
+                src = "(bool)(strlen(" + src + ") > 0 ? 1 : 0)";
```

Comment:
```suggestion
                src = "strlen(" + src + ") > 0 ? true : false)";
```

Reply:
And what about `"(bool)(strlen(" + src + ")>0)"`?